### ISSUE TITLE
Make ResponderEventPlugin cross-platform

### ISSUE THREAD

**Issue by gaearon** [facebook/react#6352]
@dieppe worked on making `ResponderEventPlugin` work with DOM in #4303 but this was put on hold.
After #6338 brings all the relevant files into React repo, we can start thinking about it again.

I’m creating this issue to keep track of whether we want this to happen or not, and what is our plan regarding this. If there is a consensus, we can look into rebasing #4303.

cc @sebmarkbage @jordwalke

**Comment by necolas** [comment]
I took #4303 and have made adjustments to provide initial web support for the Responder system. The work is spread over 3 places – injecting the plugin, normalizing native events, and intercepting events in `View` before original handlers are called. It would be great if there were eventually official DOM support for the Responder system and PanResponder

* [injectResponderEventPlugin](https://github.com/necolas/react-native-web/blob/7aa760506a799b1e2a72217c4d2299c229359329/src/modules/injectResponderEventPlugin.js)
* [normalizeNativeEvent](https://github.com/necolas/react-native-web/blob/7aa760506a799b1e2a72217c4d2299c229359329/src/modules/normalizeNativeEvent.js)
* [View event handlers](https://github.com/necolas/react-native-web/blob/7aa760506a799b1e2a72217c4d2299c229359329/src/components/View/index.js)

**Comment by sebmarkbage** [comment]
I want.

**Comment by necolas** [comment]
https://github.com/facebook/react/pull/12629 upstreamed some of the dependencies to get things working with DOM. Latest hackery beyond that is [here](https://github.com/necolas/react-native-web/blob/50442c4/packages/react-native-web/src/modules/ResponderEventPlugin/index.js#L49-L74).

**Comment by necolas** [comment]
You can use https://github.com/necolas/react-native-web if you need to use the REP on web. In the future we develop a new cross-platform system to replace the REP.
